examples/pytorch_distributed/pytorch_distributed.py

```diff
@@ -0,0 +1,81 @@
+# A basic example showing how to use Runhouse to run a PyTorch distributed training script on a cluster of GPUs.
+
+import asyncio
+
+import runhouse as rh
+import torch
+
+from torch.nn.parallel import DistributedDataParallel as DDP
+
+
+# Define the PyTorch distributed training script
+def train():
+    # Initialize the distributed training environment,
+    # per https://pytorch.org/docs/stable/distributed.html#initialization
+    torch.distributed.init_process_group(backend="nccl")
+    rank = torch.distributed.get_rank()
+    print(f"Rank {rank} of {torch.distributed.get_world_size()} initialized")
+
+    # Create a simple model and optimizer
+    device_id = rank % torch.cuda.device_count()
+    model = torch.nn.Linear(10, 1).cuda()
+    model = DDP(model, device_ids=[device_id])
+    optimizer = torch.optim.SGD(model.parameters(), lr=0.01)
+
+    # Perform a simple training loop
+    for epoch in range(10):
+        optimizer.zero_grad()
+        output = model(torch.randn(10).cuda())
+        loss = output.sum()
+        loss.backward()
+        optimizer.step()
+
+        print(f"Rank {rank}: Epoch {epoch}, Loss {loss.item()}")
+
+    # Clean up the distributed training environment
+    torch.distributed.destroy_process_group()
+
+
+async def main():
+    # Create a cluster of 2 GPUs
+    gpus_per_node = 1
+    num_nodes = 2
+    cluster = (
+        rh.cluster(
+            name=f"rh-{num_nodes}x{gpus_per_node}GPU",
+            instance_type=f"A10G:{gpus_per_node}",
+            num_instances=num_nodes,
+        )
+        .up_if_not()
+        .save()
+    )
+    train_workers = []
+    for i in range(num_nodes):
+        for j in range(gpus_per_node):
+            # Per https://pytorch.org/docs/stable/distributed.html#environment-variable-initialization
+            dist_config = {
+                "MASTER_ADDR": cluster.internal_ips[0],
+                "MASTER_PORT": "12345",
+                "RANK": str(i * gpus_per_node + j),
+                "WORLD_SIZE": str(num_nodes * gpus_per_node),
+            }
+            env = rh.env(
+                name=f"pytorch_env_{i}",
+                reqs=["torch"],
+                env_vars=dist_config,
+                compute={"node_idx": i},
+            )
+            # While iterating, you can kill the worker processes to stop any pending or handing calls
+            # cluster.delete(env.name)
+            train_worker = rh.function(train).to(cluster, env=env, name=f"train_{i}")
+            train_workers.append(train_worker)
+
+    # Call the workers async to run them concurrently (each will connect and
+    # wait for the others during init_process_group)
+    await asyncio.gather(
+        *[train_worker(run_async=True) for train_worker in train_workers]
+    )
+
+
```

Comment:
Helpful to add:
```python
# :::note{.info title="Note"}
# Make sure that your code runs within a `if __name__ == "__main__":` block, as shown below. Otherwise,
# the script code will run when Runhouse attempts to run code remotely.
# :::
```